__tests__/views/modals/ModalTransactionCosignature.spec.ts

```diff
@@ -244,4 +263,132 @@ describe('ModalTransactionCosignature', () => {
+
+            // Assert:
+            expect(component.allowUnknownMultisigTransactions).toBe(expectations.formToBeShown);
+            expect(warningElement.exists()).toBe(expectations.warningToBeShown);
+            expect(formElement.exists()).toBe(expectations.formToBeShown);
+        };
+
+        test('render form when positive', async () => {
+            // Arrange
+            const allowUnknownMultisigTransactions = true;
+            const expectations = {
+                warningToBeShown: false,
+                formToBeShown: true,
+            };
+
+            // Act + Assert:
+            await runAllowUnknownMultisigTransactionsTest(allowUnknownMultisigTransactions, expectations);
+        });
+
+        test('render warning when negative', async () => {
+            // Arrange
+            const allowUnknownMultisigTransactions = false;
+            const expectations = {
+                warningToBeShown: true,
+                formToBeShown: false,
+            };
+
+            // Act + Assert:
+            await runAllowUnknownMultisigTransactionsTest(allowUnknownMultisigTransactions, expectations);
+        });
+    });
+
+    describe('showKnownSignerAlert', () => {
+        const signerAccount = getTestAccount('cosigner1').publicAccount;
+
+        const runKnownSignerAlertTest = async (
+            addressBook: AddressBook,
+            isAddressInMultisigTree: boolean,
+            expectAlertToBeShown: boolean,
+        ) => {
+            // Arrange:
+            const aggregateTransaction = createAggregateBondedTransaction(signerAccount, currentAccount);
+            const store = createStore(aggregateTransaction, TransactionGroupEnum.Partial, addressBook);
+            jest.spyOn(MultisigService, 'isAddressInMultisigTree').mockReturnValue(isAddressInMultisigTree);
+
+            // Act:
+            const wrapper = shallowMount(ModalTransactionCosignature, {
+                localVue,
+                i18n,
+                store,
+                router,
+            });
+            const component = wrapper.vm as ModalTransactionCosignatureTs;
+            await wrapper.setProps({ transactionHash });
+            await new Promise(process.nextTick);
+            const alertElement = wrapper.find('alert-stub');
+            const isAlertRendered = !!alertElement.attributes('visible');
+
+            // Assert:
+            expect(component.showKnownSignerAlert).toBe(expectAlertToBeShown);
+            expect(isAlertRendered).toBe(expectAlertToBeShown);
+        };
+
+        test('positive when signer is a member of multisig tree', async () => {
+            // Arrange
+            const addressBook = new AddressBook([]);
+            const isAddressInMultisigTree = true;
+            const expectAlertToBeShown = true;
+
+            // Act + Assert:
+            await runKnownSignerAlertTest(addressBook, isAddressInMultisigTree, expectAlertToBeShown);
+        });
```

Comment:
question: why this behavior?

Reply:
If the account is in an address book or a member of the multisig tree, we consider that account to be trusted (or known). In that case we do not present the unknown signer **warning** panel (the puprle one). Instead we present a sign form with a small **alert** box which ask user to verify transaction before signing.